types/react-json/index.d.ts

```diff
@@ -0,0 +1,19 @@
+// Type definitions for react-json 0.2
+// Project: https://github.com/arqex/react-json
+// Definitions by: Christoph Spielmann <https://github.com/spielc>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.3
+
+import * as React from 'react';
+
+type OnChangeHandler = (value: any) => void;
+
+interface JsonProperties {
+    value: any;
+    onChange?: OnChangeHandler;
+}
+
+declare class Json extends React.Component<JsonProperties> {
+}
+
+export default Json;
```

Comment:
[This is not actually a `default` export.](https://github.com/arqex/react-json/blob/81bbeefacf9ceadbd9a7e92b9481f43608d266cd/Json.js#L177) This should be `export = Json`.

Read up more here: https://github.com/DefinitelyTyped/DefinitelyTyped/tree/92d393285cf42e347a0c687e10cd1e1a5f418a49#a-package-uses-export--but-i-prefer-to-use-default-imports-can-i-change-export--to-export-default

Reply:
Hmm now i have to admit you have finally managed to confuse me completely...

If I apply the change as you suggested (`export = Json`) then i can't use a "regular" import (`import Json from 'react-json';`) as this doesn't compile any more. If i use `import Json = require('react-json')` it compiles fine but at runtime (browser) i get a ReferenceError for the type Json. The only thing that compiles AND runs is using `const Json = require('react-json')`, but this will cause the linter to choke as this is discouraged?!? I thought i understood module loading in JS at least mostly but now I am not so sure any more... Maybe you were right with your first instinct and it makes no sense to write a declaration file for this package

Reply:
No, I'm sorry I've been terse without an explanation.

The library was authored in such a way that `export =` and `import Json = require("react-json")` is the most faithful way to declare and import the module.

If your build pipeline supports `import Json from "react-json"` then you're probably using Babel or Webpack. What you'll need to do in TypeScript is use the `allowSyntheticDefaultImports` flag to signal to TypeScript what the runtime behavior will be.

Reply:
Ah now this makes sense... I created my application using the TypeScript+React-Quick Start-Guide so your assumption with Webpack was correct and the hint with the additional compiler option finally was the "breakthrough"!

Thanks for your patience and pointing out my mistakes Daniel! I learned alot during the last two days!

Reply:
You're welcome! Sorry I didn't give you a more complete explanation sooner!